portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,16 +49,17 @@
 
   @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        ArrayList<String> comments = new ArrayList<>();
+        ArrayList<String[]>  messagesAndURLs = new ArrayList<>();
         Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
         PreparedQuery results = datastore.prepare(query);
         for (Entity entity : results.asIterable()) {
             String text = (String) entity.getProperty("text");
-            comments.add(text);
-        }
+            String imageUrl = (String) entity.getProperty("imageUrl");
 
+            messagesAndURLs.add( new String[]{text,imageUrl}  );
```

Comment:
Instead of packing information into the primitive array, and relying on order and convention, it would be preferred to create a simple class `Message` or similar to hold this information.

Something like

```java
public class Message {
  private String comment;
  private String imageUrl;

  public Message(String comment, String imageUrl) {
    this.comment = comment;
    this.imageUrl = imageUrl;
  }
}